ocs_ci/ocs/jenkins.py

```diff
@@ -0,0 +1,193 @@
+
+
+log = logging.getLogger(__name__)
+
+
+class Jenkins(object):
+    """
+    Workload operation using Jenkins
+    """
+    def __init__(self):
+        """
+        Initializer function
+
+        """
+        self.ocp = OCP()
+        self.build_completed = []
+
+    def create_ocs_jenkins_template(self):
+        """
+        Create jenkins template
+
+        """
+        try:
+            ocs_jenkins_template = templating.load_yaml(constants.JENKINS_TEMPLATE)
+            self.ocs_jenkins_template = OCS(**ocs_jenkins_template)
+            self.ocs_jenkins_template.create()
+        except(CommandFailed, CalledProcessError) as cf:
+            log.error('Failed during create ocs jenkins template')
+            raise cf
+
+    def create_namespace(self):
+        """
+        create namespace for Jenkins
+        """
+        self.ocp.new_project(constants.JENKINS_NAMESPACE)
+
+    def setup_jenkins_build_config(self):
+        """
+        Setup jenkins build config
+
+        """
+        try:
+            jenkins_build_config = templating.load_yaml(constants.JENKINS_BUILDCONFIG_YAML)
+            self.jenkins_build_config = OCS(**jenkins_build_config)
+            self.jenkins_build_config.create()
+        except(CommandFailed, CalledProcessError) as cf:
+            log.error('Failed during setup of Jenkins build config')
+            raise cf
+
+    def wait_for_jenkins_deploy_status(self, status, timeout=None):
+        """
+        Wait for jenkins deploy pods status to reach running/completed
+
+        Args:
+            status (str): status to reach Running or Completed
+            timeout (int): Time in seconds to wait
+
+        """
+        timeout = timeout if timeout else 600
+        # Wait for jenkins-deploy pods to reached and running
+        log.info(f"Waiting for jenkins-deploy pods to be reach {status} state")
+        jenkins_deploy_pods = self.get_jenkins_deploy_pods()
+        for jenkins_deploy_pod in jenkins_deploy_pods:
+            try:
+                wait_for_resource_state(
+                    resource=jenkins_deploy_pod, state=status, timeout=timeout
+                )
+            except ResourceWrongStatusException:
+                output = run_cmd(f'oc logs {jenkins_deploy_pod.name} -n {constants.JENKINS_NAMESPACE}')
+                error_msg = f'{jenkins_deploy_pod.name} did not reach to {status} state after {timeout} sec\n{output}'
+                log.error(error_msg)
+                raise UnexpectedBehaviour(error_msg)
+
+    def wait_for_build_status(self, status, timeout=None):
+        """
+        Wait for jenkins deploy pods status to reach running/completed
+
+        Args:
+            status (str): status to reach Running or Completed
+            timeout (int): Time in seconds to wait
+
+        """
+        timeout = timeout if timeout else 600
+        # Wait for jenkins-deploy pods to reached and running
```

Comment:
Here we are waiting for the build status.. Please update the comment accordingly.